src/rendering/vcLineRenderer.cpp

```diff
@@ -99,62 +100,87 @@ udResult vcLineRenderer_CreateLine(vcLineInstance **ppLine)
 
 udResult vcLineRenderer_UpdatePoints(vcLineInstance *pLine, const udDouble3 *pPoints, size_t pointCount, const udFloat4 &colour, float width, bool closed)
 {
+  // Cleanup previous state
+  vcMesh_Destroy(&pLine->pMesh);
+
+  if (pointCount < 2)
+    return udR_Success;
+
   udResult result;
 
-  closed = closed && (pointCount > 2); // Less than 2 points can't really be 'closed'
-  size_t realPointCount = (pointCount + (closed ? 1 : 0));
+  size_t segmentCount = (pointCount - (closed ? 0 : 1));
 
-  size_t vertCount = realPointCount * 2;
+  //The basic idea is to break the line up into segments.
+  size_t vertCount = segmentCount * 4;
   vcLineVertex *pVerts = nullptr;
   vcMesh *pMesh = nullptr;
   udDouble3 origin = pPoints[0];
-
-  // Cleanup previous state
-  vcMesh_Destroy(&pLine->pMesh);
-
+  size_t indexCount = segmentCount * 6;
+  int *pIndices = nullptr;
+  size_t vertIndex = 0;
+  size_t indIndex = 0;
+  
   pVerts = udAllocType(vcLineVertex, vertCount, udAF_None);
   UD_ERROR_NULL(pVerts, udR_MemoryAllocationFailure);
+  
+  pIndices = udAllocType(int, indexCount, udAF_None);
+  UD_ERROR_NULL(pIndices, udR_MemoryAllocationFailure);
 
-  for (size_t i = 0; i < realPointCount; ++i)
+  for (size_t i = 0; i < segmentCount; i++)
   {
-    udDouble3 p = pPoints[i % pointCount];
-
-    udDouble3 prev = udDouble3::zero();
-    udDouble3 next = udDouble3::zero();
-    if (i == 0 && !closed)
-      prev = p + udNormalize3(p - pPoints[i + 1]) * 0.1; // fake a vert
-    else
-      prev = pPoints[(i - 1 + pointCount) % pointCount];
-
-    if (i == pointCount - 1 && !closed)
-      next = p + udNormalize3(p - pPoints[i - 1]) * 0.1; // fake a vert
-    else
-      next = pPoints[(i + 1) % pointCount];
-
-    size_t curIndex = i * 2;
-    udFloat3 p0 = udFloat3::create(p - origin);
-    udFloat3 prev0 = udFloat3::create(prev - origin);
-    udFloat3 next0 = udFloat3::create(next - origin);
-    pVerts[curIndex + 0].position = udFloat4::create(p0, -1.0f);
-    pVerts[curIndex + 1].position = udFloat4::create(p0, 1.0f);
-    pVerts[curIndex + 0].previous = udFloat4::create(prev0, -1.0f);
-    pVerts[curIndex + 1].previous = udFloat4::create(prev0, 1.0f);
-    pVerts[curIndex + 0].next = udFloat4::create(next0, -1.0f);
-    pVerts[curIndex + 1].next = udFloat4::create(next0, 1.0f);
+    //Verts
+    udDouble3 p0 = pPoints[i] - origin;
+    udDouble3 p1 = pPoints[(i + 1) % pointCount] - origin;
+    udDouble4 p0_neighbour = udDouble4::create(pPoints[(i + pointCount - 1) % pointCount] - origin, 1.0);
+    udDouble4 p1_neighbour = udDouble4::create(pPoints[(i + 2) % pointCount] - origin, 1.0);
+
+    if (i == 0 && closed)
+      p0_neighbour.w = 0.0; //Flag endpoint with w == 0
+
+    if (i == pointCount - 2 && closed)
+        p0_neighbour.w = 0.0; //Flag endpoint with w == 0
```

Comment:
Can I get a confirm that these are supposed to be `closed`? This behaviour makes more intuitive sense as `!closed`?

Reply:
Yes, these should be `!closed`